index.js

```diff
@@ -88,27 +88,54 @@ NodePreGypGithub.prototype.uploadAsset = function(cfg){
 		filePath: cfg.filePath
 	}, function(err){
 		if(err) {console.error(err); return;}
-		console.log('Staged file ' + cfg.fileName + ' saved to ' + this.owner + '/' +  this.repo + ' release ' + this.package_json.version + ' successfully.');
+		console.log('Staged file ' + cfg.fileName + ' saved to ' + this.owner + '/' +  this.repo + ' release ' + this.release.tag_name + ' successfully.');
 	}.bind(this));
 };
 
-NodePreGypGithub.prototype.uploadAssets = function(){
-	var asset;
-	console.log("Stage directory path: " + path.join(this.stage_dir));
-	fs.readdir(path.join(this.stage_dir), function(err, files){
-		if(typeof files === 'undefined') {console.log('no files found'); return;}
-		files.forEach(function(file){
+function asyncFlatMap(list, iterator, callback) {
+	var result = [], left = list.length;
+	if (!left) return callback(result);
+	list.forEach(function(item) {
+		iterator(item, function(item) {
+			if (item != null) result = result.concat(item);
+			if (!--left) callback(result);
+		});
+	});
+}
+
+function findFilesDeep(dir, callback) {
+	fs.readdir(dir, function(err, names) {
+		asyncFlatMap(names || [], function(name, next) {
+			var file = path.join(dir, name);
+			fs.stat(file, function(err, stat) {
+				if (stat) {
+					if (stat.isFile()) { return next(file); }
+					else if (stat.isDirectory()) { return findFilesDeep(file, next); }
+				}
+				next();
+			});
+		}, callback);
+	});
+};
+
+NodePreGypGithub.prototype.uploadAssets = function() {
+	console.log("Stage directory path: " + this.stage_dir);
+	findFilesDeep(this.stage_dir, function(paths) {
```

Comment:
Maybe I wasn't clear at first, but when one adds "remote_path" option to "binary" in package.json - `node-pre-gyp` puts a package into `stage_dir` / `remote_path`. So when "remote_path" == "{version}" the `stage_dir` becomes `build/stage/1.0.0/{file}.tgz` instead of `build/stage/{file}.tgz`.

We can only wild guess where the file should be since we can't use `path.join(stage_dir, package_json.remote_path)`.
Which leaves us with the options:
1. `stage_dir / tag_name` where `tag_name` defaults to `package_json.version` or is from --tag option (when remote_path is set and not empty)
2. `stage_dir`  otherwise (when remote_path is empty)

Yes, we could look for files only in those 2. But unfortunately they are nested in each other which could lead to all kind of corner cases. (like existing directory `stage_dir/tag_name` but without files etc.). So I thought that a generic solution would be better since I did not had to analyze all possible matrixes of options to guess right every time. I just expect some files to be there. Anywhere below `stage_dir`. No wild-guessing just wild-carding.